webapp/src/components/InputField/index.tsx

```diff
@@ -1,70 +1,52 @@
-import * as React from 'react';
-import {FormattedMessage} from 'react-intl';
+import React, {ReactNode} from 'react';
 
 import {InputFieldType} from 'types';
 
 type Props = {
-    heading: {
-        id: string;
-        message: string;
-    };
+    heading: ReactNode;
     tagType: string;
-    input: {
-        type?: string;
-        placeholder?: string;
-        maxLength?: number;
-        min?: number;
-        rows?: number;
-        defaultValue?: number;
-        onChange: (e: React.ChangeEvent<InputFieldType>) => void;
-        value: string | number;
-        disabled?: boolean;
-    };
-    description: {
-        id: string;
-        message: string;
-    }
+    type?: string;
```

Comment:
```suggestion
    type?: 'text' | 'number' | 'textarea';
```
Above is just an example. You should create an enum for this.